python-checks/src/test/resources/checks/hashMethod.py

```diff
@@ -0,0 +1,39 @@
+class EmptyClass:
+    ...
+
+class ClassWithEq:
+    def __eq__(self, other):
+        return self
+
+class ClassWithHash:
+    def __hash__(self):
+        return 42
+
+class ClassWithEqInherithingHash(ClassWithHash):
+    def __eq__(self, other):
+        return self
+
+
+def set_members():
+    {EmptyClass()}  # OK - object has by default a __hash__() method
+    {ClassWithEq()} # FN
+    {ClassWithEqInherithingHash()} # FN
+    my_list = [1, 2, 3]
+    {my_list} # Noncompliant {{Make sure this expression is hashable.}}
+#    ^^^^^^^
+    {1, 2, 3}
+    {ClassWithHash()}
+    {some_func()}
+    {None}
+
+def dictionary_keys():
+    {EmptyClass(): 42} # OK
+    my_list = [1, 2, 3]
+    {my_list: 42} # Noncompliant
+    {1: "one", 2: "two"}
+    {ClassWithHash(): 42}
+    {None: 42}
```

Comment:
Minor: Maybe we should add some cases involving dictionary comprehension / set comprehension.
E.g.:

```python
{k: 42 for k in [1, 2]}
sequence_of_unhashable: Sequence[List] = [[], []]
{k: 42 for k in sequence_of_unhashable} # Noncompliant (?) - Example 1
{[]: v for v in [1, 2]} # Noncompliant - Example 2
```

Although, I am not sure if our tooling is currently sufficient for inferring the type of the key expression in many cases.
I did a quick experiment where I inspected a `DictCompExpression` generated for Example 1, and it was not able to infer the type of k, even when I added the type annotations.

For Example 2 it is able to determine the type.

Reply:
Ah, also: The key value expressions in dictionary comprehensions are not parsed as KEY_VALUE_PAIR nodes. Hence, they are currently not checked by the implementation.
Instead, we have to register a consumer for `DICT_COMPREHENSION`. The same probably applies for set comprehensions.

Reply:
Thanks, good point! I will create a separate ticket in the backlog to handle dict and set comprehensions!